File: app/src/test/java/com/starsep/sokoban/release/activity/MainMenuActivityTest.java
package com.starsep.sokoban.release.activity;

import android.content.Intent;
import android.widget.Button;

import com.starsep.sokoban.release.BuildConfig;
import com.starsep.sokoban.release.R;
import com.starsep.sokoban.release.activity.MainMenuActivity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;
import org.robolectric.shadows.ShadowAlertDialog;

import static android.view.View.VISIBLE;
import static org.junit.Assert.assertEquals;
import static org.robolectric.Shadows.shadowOf;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class)
public class MainMenuActivityTest {
    private MainMenuActivity activity;

    @Before
    public void setUp() throws Exception {
        activity = Robolectric.buildActivity(MainMenuActivity.class)
                .create()
                .resume()
                .get();
    }

    @Test
    public void shouldHaveMenuButtons() {
        Button newGameButton = activity.findViewById(R.id.newGameButton);
        assertEquals(VISIBLE, newGameButton.getVisibility());
        Button helpButton = activity.findViewById(R.id.helpButton);
        assertEquals(VISIBLE, helpButton.getVisibility());
        Button settingsButton = activity.findViewById(R.id.settingsButton);
        assertEquals(VISIBLE, settingsButton.getVisibility());
    }

    @Test
    public void newGameButtonLaunchesChooseLevelActivity() {
        Button newGameButton = activity.findViewById(R.id.newGameButton);
        newGameButton.performClick();
        Intent intent = shadowOf(activity).peekNextStartedActivity();
        assertEquals(ChooseLevelActivity.class.getCanonicalName(), intent.getComponent().getClassName());
    }

    @Test
    public void helpButtonShowsDialog() {
        Button helpButton = activity.findViewById(R.id.helpButton);
        helpButton.performClick();
        ShadowAlertDialog helpDialog = shadowOf(ShadowAlertDialog.getLatestAlertDialog());
        assertEquals(activity.getText(R.string.help_msg), helpDialog.getMessage());
        assertEquals(activity.getText(R.string.help), helpDialog.getTitle());
    }
}

File: app/src/test/java/com/starsep/sokoban/release/activity/MainMenuActivityTest.kt
package com.starsep.sokoban.release.activity

// @RunWith(RobolectricTestRunner::class)
class MainMenuActivityTest : RobolectricTest() {
    // TODO: fix
    /*private var activity: MainActivity? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        activity = Robolectric.buildActivity(MainActivity::class.java)
            .create()
            .resume()
            .get()
    }

    @After
    fun after() {
        close()
    }

    @Test
    fun shouldHaveMenuButtons() {
        val newGameButton =
            activity!!.findViewById<Button>(R.id.newGameButton)
        Assert.assertEquals(
            View.VISIBLE.toLong(),
            newGameButton.visibility.toLong()
        )
        val helpButton =
            activity!!.findViewById<Button>(R.id.helpButton)
        Assert.assertEquals(
            View.VISIBLE.toLong(),
            helpButton.visibility.toLong()
        )
        val settingsButton =
            activity!!.findViewById<Button>(R.id.settingsButton)
        Assert.assertEquals(
            View.VISIBLE.toLong(),
            settingsButton.visibility.toLong()
        )
    }

    @Test
    fun newGameButtonLaunchesChooseLevelActivity() {
        val newGameButton =
            activity!!.findViewById<Button>(R.id.newGameButton)
        newGameButton.performClick()
        val intent = Shadows.shadowOf(activity).peekNextStartedActivity()
        Assert.assertEquals(
            ChooseLevelFragment::class.java.canonicalName,
            intent.component!!.className
        )
    }

    @Test
    fun helpButtonShowsDialog() {
        val helpButton =
            activity!!.findViewById<Button>(R.id.helpButton)
        helpButton.performClick()
        val helpDialog =
            Shadows.shadowOf(ShadowAlertDialog.getLatestAlertDialog())
        Assert.assertEquals(activity!!.getText(R.string.help_msg), helpDialog.message)
        Assert.assertEquals(activity!!.getText(R.string.help), helpDialog.title)
    }*/
}
